Mathlib/CategoryTheory/Grothendieck.lean

```diff
@@ -149,6 +161,100 @@ def forget : Grothendieck F ⥤ C where
 
 end
 
+section
+
+variable {G : C ⥤ Cat}
+
+/-- The Grothendieck construction is functorial: a natural transformation `α : F ⟶ G` induces
+a functor `Grothendieck.map : Grothendieck F ⥤ Grothendieck G`.
+-/
+@[simps!]
+def map (α : F ⟶ G) : Grothendieck F ⥤ Grothendieck G where
+  obj X := {
+    base := X.base
+    fiber := (α.app X.base).obj X.fiber }
+  map {X Y} f := {
+    base := f.base
+    fiber := (eqToHom (α.naturality f.base).symm).app X.fiber ≫ (α.app Y.base).map f.fiber }
+  map_id X := by
+    simp only [Cat.comp_obj, id_fiber', eqToHom_map]
+    congr 1
+    rw [eqToHom_app, eqToHom_trans]
+  map_comp {X Y Z} f g := by
+    dsimp
+    congr 1
+    simp only [comp_fiber' f g, ← Category.assoc, Functor.map_comp, eqToHom_map]
+    apply congrFun (congrArg (. ≫ .) ?_) ((α.app Z.base).map g.fiber)
+    have H := Functor.congr_hom (α.naturality g.base).symm f.fiber
+    erw [H, eqToHom_app, eqToHom_app]
+    simp only [Cat.comp_obj, eqToHom_trans, eqToHom_map, Cat.comp_map, eqToHom_trans_assoc,
+      Category.assoc]
+    erw [eqToHom_app]
+    simp
+
+theorem map_obj {α : F ⟶ G} (X : Grothendieck F) :
+    (Grothendieck.map α).obj X = ⟨X.base, (α.app X.base).obj X.fiber⟩ := by
+  rfl
+
+theorem map_map {α : F ⟶ G} {X Y : Grothendieck F} {f : X ⟶ Y} :
+    (Grothendieck.map α).map f =
+    ⟨f.base, (eqToHom (α.naturality f.base).symm).app X.fiber ≫ (α.app Y.base).map f.fiber⟩ := by
+  rfl
+
+/-- The functor `Grothendieck.map α : Grothendieck F ⥤ Grothendieck G` lies over `C`.-/
+theorem functor_comp_forget {α : F ⟶ G} :
+    Grothendieck.map α ⋙ Grothendieck.forget G = Grothendieck.forget F := by
+  rfl
+
+theorem map_id_eq : map (𝟙 F) = 𝟙 (Cat.of <| Grothendieck <| F) := by
+  fapply Functor.ext
+  · intro X
+    rfl
+  · intro X Y f
+    simp [map_map]
+    congr
+    rw [NatTrans.id_app]
+    simp
+
+/-- Making the equality of functors into an isomorphism. Note: we should avoid equality of functors
+if possible, and we should prefer `map_id_iso` to `map_id_eq` whenever we can. -/
+def mapIdIso : map (𝟙 F) ≅ 𝟙 (Cat.of <| Grothendieck <| F) := eqToIso map_id_eq
+
+variable {H : C ⥤ Cat}
+
+theorem map_comp_eq (α : F ⟶ G) (β : G ⟶ H) :
+    map (α ≫ β) = map α ⋙ map β := by
+  fapply Functor.ext
+  · intro X
+    rfl
+  · intro X Y f
+    simp [map_map]
+    fapply Grothendieck.ext
+    · rfl
+    · simp
```

Comment:
Missed these non-terminal simps. please squeeze them.

Reply:
replaces `simp` proofs by `simp only [...]` and minimized the arguments of `simp only`.